source/addonHandler/__init__.py

```diff
@@ -434,18 +434,20 @@ def completeInstall(self) -> str:
 
 	def requestRemove(self):
 		"""Marks this addon for removal on NVDA restart."""
-		if self.isPendingInstall:
+		if self.isPendingInstall and not self.isInstalled:
+			# Handle removal of an add-on not yet installed
 			self.completeRemove()
 			state[AddonStateCategory.PENDING_INSTALL].discard(self.name)
 			state[AddonStateCategory.OVERRIDE_COMPATIBILITY].discard(self.name)
-			#Force availableAddons to be updated
+			# Force availableAddons to be updated
 			getAvailableAddons(refresh=True)
 		else:
+			# Upgrade to existing add-on or installation of new add-on
 			state[AddonStateCategory.PENDING_REMOVE].add(self.name)
 			# There's no point keeping a record of this add-on pending being disabled now.
 			# However, if the addon is disabled, then it needs to remain disabled so that
 			# the status in add-on store continues to say "disabled"
-			state[AddonStateCategory.PENDING_INSTALL].discard(self.name)
+			state[AddonStateCategory.PENDING_DISABLE].discard(self.name)

```

Comment:
this change fixes a bug brought it when updating the `AddonStateCategory` values